components/expenses/ExpenseItemForm.js

```diff
@@ -118,6 +122,37 @@ const AttachmentLabel = () => (
   </Span>
 );
 
+const WithOCRComparisonWarning = ({ comparison, formatValue, children, mrClass = 'mr-10' }) => (
+  <div className="relative">
+    {children}
+    {Boolean(comparison?.hasMismatch) && (
+      <div className={cn('absolute right-0 top-0 mt-[9px]', mrClass)}>
+        <Tooltip>
+          <TooltipTrigger>
+            <AlertTriangle size={16} color="#CB9C03" />
+          </TooltipTrigger>
+          <TooltipContent>
+            <FormattedMessage
+              defaultMessage="This value does not match the one scanned from the document ({value})"
+              values={{ value: formatValue ? formatValue(comparison.ocrValue) : comparison.ocrValue.toString() }}
+            />
+          </TooltipContent>
+        </Tooltip>
+      </div>
+    )}
+  </div>
+);
```

Comment:
Not a big fan of HOC and absolute positioning but I understand this is a bit of a tricky place to render an alert icon.
Any chance we could use the input itself to render some sort of yellow error state?

Reply:
The absolute positioning will be difficult to avoid since we want the icon inside the input.

I initially wanted to implement that on `StyledInputField` but backed out of this approach because it was difficult to develop an API that would stay compatible with all the existing props (`hint`, `padding`, etc...).
This `WithOCRComparisonWarning` was a way to introduce the feature with minimal diffs and a no side-effects guarantee.

I'm considering returning to this approach as we'll try to resolve https://github.com/opencollective/opencollective/issues/7039.

Reply:
Agreed! Maybe we can find an easier way to do that with Radix-UI components.